### ISSUE TITLE
Batch Normalization not updating moving mean for GAN discriminator. This results in huge test losses.  

### ISSUE THREAD

**Issue by gulrukh-k** [keras-team/keras#12353]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by gulrukh-k** [comment]
The problem can be solved by setting discriminator layers to not trainable instead of the model. There are two logs printing BN weights corresponding to above two versions. The problem is quite clear.
[disc_trainable.log](https://github.com/keras-team/keras/files/2909566/disc_trainable.log)
[layer_trainable.log](https://github.com/keras-team/keras/files/2909568/layer_trainable.log)

**Comment by bigmao8576** [comment]
Thank you for sharing. Would you please offer more info? I tried the following code, but the moving mean and std are still not updating:
`for i in range(len(disc.layers)):
    disc.layers[i].trainable = False`

Keras version 2.3.1
tf version 1.15.0
